pkg/resources/management.cattle.io/v3/projectroletemplatebinding/validator.go

```diff
@@ -141,50 +145,60 @@ func clusterFromProject(project string) (string, string) {
-func validateUpdateFields(oldPRTB, newPRTB *apisv3.ProjectRoleTemplateBinding) error {
-	var invalidFieldName string
+func validateUpdateFields(oldPRTB, newPRTB *apisv3.ProjectRoleTemplateBinding, fieldPath *field.Path) *field.Error {
 	switch {
 	case oldPRTB.RoleTemplateName != newPRTB.RoleTemplateName:
-		invalidFieldName = "referenced roleTemplate"
+		return field.Forbidden(fieldPath.Child("roleTemplateName"), "")
 	case oldPRTB.ProjectName != newPRTB.ProjectName:
-		invalidFieldName = "projectName"
+		return field.Forbidden(fieldPath.Child("projectName"), "")
 	case oldPRTB.UserName != newPRTB.UserName && oldPRTB.UserName != "":
-		invalidFieldName = "userName"
+		return field.Forbidden(fieldPath.Child("userName"), "")
 	case oldPRTB.UserPrincipalName != newPRTB.UserPrincipalName && oldPRTB.UserPrincipalName != "":
-		invalidFieldName = "userPrincipalName"
+		return field.Forbidden(fieldPath.Child("userPrincipalName"), "")
 	case oldPRTB.GroupName != newPRTB.GroupName && oldPRTB.GroupName != "":
-		invalidFieldName = "groupName"
+		return field.Forbidden(fieldPath.Child("groupName"), "")
 	case oldPRTB.GroupPrincipalName != newPRTB.GroupPrincipalName && oldPRTB.GroupPrincipalName != "":
-		invalidFieldName = "groupPrincipalName"
+		return field.Forbidden(fieldPath.Child("groupPrincipalName"), "")
 	case (newPRTB.GroupName != "" || oldPRTB.GroupPrincipalName != "") && (newPRTB.UserName != "" || oldPRTB.UserPrincipalName != ""):
-		invalidFieldName = "both user and group"
+		return field.Forbidden(fieldPath.Child("groupName", "groupPrincipalName", "userName", "userPrincipalName"), "")
+	case newPRTB.ServiceAccount != "":
+		return field.Forbidden(fieldPath.Child("serviceAccount"), "")
 	default:
 		return nil
 	}
-
-	return fmt.Errorf("cannot update %s for clusterRoleTemplateBinding %s: %w", invalidFieldName, oldPRTB.Name, admission.ErrInvalidRequest)
 }
 
 // validateCreateFields checks if all required fields are present and valid.
-func (v *Validator) validateCreateFields(newPRTB *apisv3.ProjectRoleTemplateBinding) error {
+func (v *Validator) validateCreateFields(newPRTB *apisv3.ProjectRoleTemplateBinding, fieldPath *field.Path) *field.Error {
+	if newPRTB.ServiceAccount != "" {
+		return field.Forbidden(fieldPath.Child("serviceAccount"), "")
+	}
+
 	hasUserTarget := newPRTB.UserName != "" || newPRTB.UserPrincipalName != ""
 	hasGroupTarget := newPRTB.GroupName != "" || newPRTB.GroupPrincipalName != ""
 
 	if (hasUserTarget && hasGroupTarget) || (!hasUserTarget && !hasGroupTarget) {
-		return fmt.Errorf("binding must target either a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]: %w", admission.ErrInvalidRequest)
+		return field.Invalid(fieldPath.Child("groupName", "groupPrincipalName", "userName", "userPrincipalName"),
+			"combination of [userId]/[userPrincipalId] and [groupId]/[groupPrincipalId]",
+			"binding must target either a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]")
```

Comment:
If you are validating a combination of fields we can just use the parent as the fieldPath
```suggestion
		return field.Forbidden(fieldPath, "binding must target either a user [userId]/[userPrincipalId] OR a group [groupId]/[groupPrincipalId]")
```